### ISSUE TITLE
Does Backup but doesn't save to remote server

### ISSUE THREAD

**Issue by ClearlyDazed** [ikifar2012/remote-backup-addon#37]
Have used this add-on for some time but the recent update gives an error in the logs.
"[info] Backup created: 35c9e120
[warn] Copying 35c9e120.tar to /data/Documents/BACKUPS/Home_Assistant/Odroid on 192.168.1.109 using SCP
Warning: Permanently added '192.168.1.109' (ECDSA) to the list of known hosts.
Permission denied, please try again."

Last working backup was on Dec 23, 2021
Has something changed for ssh key in add-on?
I can ssh into 192.168.1.109 from HA terminal.

**Comment by ikifar2012** [comment]
You are storing the SSH key in a file in the SSL directory correct?

**Comment by ClearlyDazed** [comment]
Yes, the previous version of the add-on was working until Dec 23 , after which it was automatically updated. Redid the ssh-keygen again with the ssh key placed in the SSl directory, but still received error message. Could ssh and scp into remote server from HA terminal.

**Comment by ikifar2012** [comment]
Strange, nothing has changed. Upon running the addon a second time do you get the same error?

**Comment by ClearlyDazed** [comment]
I have just run it again and have included the log for the add-on below.
I'll try to downgrade the addon version and take a look at the remote server again.


"[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Remote Backup
 Automatically create and backup HA backups using SCP
-----------------------------------------------------------
 Add-on version: 2021.12.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 6.6  (armv7 / odroid-xu)
 Home Assistant Core: 2021.12.6
 Home Assistant Supervisor: 2021.12.2
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[info] Adding SSH key
[info] SSH key added
[warn] Creating local backup: "Automated backup 2021-12-28 11-32"
[info] Backup created: 0df02032
[warn] Copying 0df02032.tar to /data/Documents/BACKUPS/Home_Assistant/Odroid on 192.168.1.109 using SCP
Warning: Permanently added '192.168.1.109' (ECDSA) to the list of known hosts.
Permission denied, please try again.
Permission denied, please try again.
root@192.168.1.109: Permission denied (publickey,password).
lost connection
[cmd] /run.sh exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] 99-message.sh: executing... 
-----------------------------------------------------------
                Oops! Something went wrong.

 We are so sorry, but something went terribly wrong when
 starting or running this add-on.
 
 Be sure to check the log above, line by line, for hints.
-----------------------------------------------------------
[cont-finish.d] 99-message.sh: exited 0.
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal."

**Comment by ClearlyDazed** [comment]
I downgrade to Remote Backup 2012.11.2. The log below shows that the backup was coped to the remote server.

I updated Remote Backup to 2012.12.0. The log is the same as the previous post, reporting scp did not work. 

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Remote Backup
 Automatically create and backup HA backups using SCP
-----------------------------------------------------------
 Add-on version: 2021.11.2
 There is an update available for this add-on!
 Latest add-on version: 2021.12.0
 Please consider upgrading as soon as possible.
 System: Home Assistant OS 6.6  (armv7 / odroid-xu)
 Home Assistant Core: 2021.12.4
 Home Assistant Supervisor: 2021.12.2
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[info] Adding SSH key
[info] SSH key added
[warn] Creating local backup: "Automated backup 2021-12-28 11-51"
[info] Backup created: 4cd609f8
[warn] Copying 4cd609f8.tar to /data/Documents/BACKUPS/Home_Assistant/Odroid on 192.168.1.109 using SCP
Warning: Permanently added '192.168.1.109' (ECDSA) to the list of known hosts.
[info] Backup copied to /data/Documents/BACKUPS/Home_Assistant/Odroid/4cd609f8.tar on 192.168.1.109
[warn] Renaming 4cd609f8.tar to Automated backup 2021-12-28 11-51.tar
[info] Backup renamed to /data/Documents/BACKUPS/Home_Assistant/Odroid/Automated backup 2021-12-28 11-51.tar on 192.168.1.109
[info] SCP complete

**Comment by ClearlyDazed** [comment]
Currently have Remote Backup 2012.11.2 working under HA Core 2021.12.6

**Comment by ikifar2012** [comment]
Both me and @SirGoodenough have attempted to recrate your issue and can't, the only change I made was to upgrade the base image and that should have had no effect on SSH. Are you sure you have allowed the root user to SSH? Is the SSH server up to date? What command did you use to generate the key? If its old I'd highly recommend creating an Ed25519 key

If you would like to discuss this further, I invite you to join the [What are we fixing today? Discord Server](https://discord.gg/Uhmhu3B)

**Comment by ClearlyDazed** [comment]
Thank you. I'll follow your recommendations.

**Comment by Kreisverkehr** [comment]
I have a similar issue. My backup stopped working on the exact same day, probably due to the update. I noticed this now because my HA Instance ran out of disk space because of that error.
It seems that an update to scp was shipped with the base image, that removed support for older key types and my old, prebuilt nas that doesn't get any updates anymore, does not offer any key type that this version of scp can understand.

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Remote Backup
 Automatically create and backup HA backups using SCP
-----------------------------------------------------------
 Add-on version: 2022.1.1
 You are running the latest version of this add-on.
 System: Home Assistant OS 7.2  (armv7 / raspberrypi4)
 Home Assistant Core: 2022.2.2
 Home Assistant Supervisor: 2022.01.1
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[info] Adding SSH key
[info] SSH key added
[warn] Creating local backup: "Automated backup 2022-02-07 00-00"
[info] Backup created: d613d6b1
[warn] Copying d613d6b1.tar to /nfs/Sicherungen/HomeAssistant on wdmycloud.local using SCP
Unable to negotiate with 192.168.2.1 port 22: no matching host key type found. Their offer: ssh-rsa,ssh-dss
lost connection
[cmd] /run.sh exited 1
[cont-finish.d] executing container finish scripts...
[cont-finish.d] 99-message.sh: executing... 
-----------------------------------------------------------
                Oops! Something went wrong.

 We are so sorry, but something went terribly wrong when
 starting or running this add-on.
 
 Be sure to check the log above, line by line, for hints.
-----------------------------------------------------------
[cont-finish.d] 99-message.sh: exited 0.
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
```

**Comment by ikifar2012** [comment]
@Kreisverkehr I really want to try to recreate this, do you know what type of key you created and how you generated them? The model of the nas would also be great

**Comment by Kreisverkehr** [comment]
@ikifar2012 Your're right. I should have included that in my first message.

**My NAS:**
**Device**: WD My Cloud 4TB
**Firmwareversion**: WDMyCloud v04.05.00-342 : Core F/W
**Linux Kernel**: WDMyCloud 3.2.26
**SSH Version**: OpenSSH_6.0p1 Debian-4, OpenSSL 1.0.1t  3 May 2016

**My Key**
**Type**: RSA
**Generated on**: Windows 10 using the builtin ssh-keygen tool (must have been 20H2 or 21H1)

I just noticed, while gathering the information that there is an update to my NAS Firmware. The changelog does not talk about SSH but I'll try updating it.

**Comment by ikifar2012** [comment]
Found this: https://askubuntu.com/questions/836048/ssh-returns-no-matching-host-key-type-found-their-offer-ssh-dss

Not sure the best way to implement adding that into the addon either I could make `HostKeyAlgorithms` an optional parameter in the addon config

**Comment by Kreisverkehr** [comment]
This seems to be resolving at least my issue. 
An optional parameter would be geat. Maybe with a warning on enabeling insecure algorithms. If the user is using this addon with a now insecure alogrithm he schould be aware of it an enable it explicitly.

**Comment by BioDuke** [comment]
> You are storing the SSH key in a file in the SSL directory correct?

TLDR:
`/usr/share/hassio/ssl`

I'm a relative noob with containers and HA.  This stumped me for a while.  I knew I was looking for the SSL directory. but it took me ages to find the correct one.  There are multiple SSL directories on my RPI HA supervised instance.  There are also multiple containers running on the machine, many had their own SSL directories.

I finally figured it out by inspecting the the remote_backup docker instance

`docker inspect addon_3490a758_remote_backup`

and found this in the config
```
...
"HostConfig": {
            "Binds": [
                "/dev:/dev:ro",
                "/usr/share/hassio/addons/data/3490a758_remote_backup:/data:rw",
                "/usr/share/hassio/homeassistant:/config:ro",
                "/usr/share/hassio/ssl:/ssl:ro",
                "/usr/share/hassio/addons/local:/addons:ro",
                "/usr/share/hassio/backup:/backup:rw",
                "/usr/share/hassio/share:/share:ro"
            ],
```
/usr/share/hassio/ssl is a directory created/used by the Homeassistant container, but the file is located on your linux instance.
